controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -234,3 +238,13 @@ func (c *Client) Alarms(ctx context.Context) ([]MemberAlarm, error) {
 
 	return memberAlarms, nil
 }
+
+// HealthCheck checks the healthiness of endpoints specified in endpoints during Client creation.
+// Using the same logic used in etcdctl health command.
+func (c *Client) HealthCheck(ctx context.Context) error {
+	_, err := c.KVClient.Get(ctx, "health")
+	if err == nil || err == rpctypes.ErrPermissionDenied {
```

Comment:
Why are we returning nil if the permission was denied? And in which cases could that happen? Let's document this in a comment right before returning

Reply:
Reason is: `permission denied is OK since proposal goes through consensus to get it`
Added as command.
I am using the same logic with etcdctl here.